### ISSUE TITLE
🚧 Universal Language Model Fine-tuning for Text Classification

### ISSUE THREAD

**Issue by fury00812** [fury00812/Survey#2]
Universal Language Model Fine-tuning for Text Classification
Jeremy Howard, Sebastian Ruder
ACL2018
https://arxiv.org/pdf/1801.06146v5.pdf

## 途中メモ
- 結構面白そう
- 3は論文の肝なので読み込む
- [素晴らしい解説](https://qiita.com/kogecoo/items/49dd9019bf9268bd8e38)
- 次 : Low Resource Text Classification with ULMFit and Backtranslation

# Abstract
- 転移学習（transfer learning）は画像認識研究に大きな衝撃を与えた。しかしNLPにおいては依然としてタスク毎の学習が必要である。
- 本稿ではNLPに適用可能な転移学習モデル"**Universal Language Model Fine-tuning(ULMFiT)**"を提案し、また要点の解説を行う。
- ULMFiTは、6つのテキスト分類タスクにおいてSotAを大きく上回り、18-24%のエラーを減少した。
- 100個のデータによって学習したULMFiTが、100個以上のデータを学習した既存モデルに匹敵する性能を示した。
- モデルは[オープンソース](http://nlp.fast.ai/ulmfit)で公開。

# 3 Universal Language Model Fine-tuning
#### Figure1 : ULMFiTの3ステップ

![スクリーンショット 2019-04-22 17 55 12](https://user-images.githubusercontent.com/35480446/59026003-728ca980-8890-11e9-8cee-5d62ee81cade.png)

> (a) 言語モデルの事前学習(3.1)
> (b) 言語モデルのfine-tuning(3.2)
> (c) 分類タスクのfine-tuning (3.3)

## 3.1 General-dmain LM pretraining
- 大規模なGeneral-domain corpusで事前学習
- 大規模かつ言語の一般的な特徴を捉えることのできるコーパスが望ましい
- 28925記事、1億300万単語からなるコーパス**Wikitext-103**を使用

## 3.2 Target task LM fine-tuning
- 3.1をターゲットタスクのコーパスによってfine-tuning
- 2つの手法:Discriminative fine-tuning, Slanted triangular learning rate
## 3.3 Target task classifier fine-tuning

# 6 Discussion and future directions

# 7 Conclusion
- NLPに適用可能な転移学習モデルを提案した。
- 同時に、幅広いタスクに渡りロバストな学習を可能にするfine-tuning手法を複数提案した。
- 6つの分類タスクにおいてSotAに比べて18-24%、error rateの低減を達成した。
